metadata/src/main/java/org/apache/kafka/controller/QuorumController.java

```diff
@@ -2314,11 +2315,24 @@ public CompletableFuture<UpdateFeaturesResponseData> updateFeatures(
         }).thenApply(result -> {
             UpdateFeaturesResponseData responseData = new UpdateFeaturesResponseData();
             responseData.setResults(new UpdateFeaturesResponseData.UpdatableFeatureResultCollection(result.size()));
-            result.forEach((featureName, error) -> responseData.results().add(
-                new UpdateFeaturesResponseData.UpdatableFeatureResult()
-                    .setFeature(featureName)
-                    .setErrorCode(error.error().code())
-                    .setErrorMessage(error.message())));
+            Optional<Entry<String, ApiError>> errorEntry = result.entrySet().stream().filter(entry ->
```

Comment:
In the previous line, we don't want to call `responseData.setResults` if there is an error, right?

Reply:
It doesn't hurt anything, but I can move it to the else if path